### ISSUE TITLE
Support to notifications

### ISSUE THREAD

**Issue by Plasmmer** [HelloZeroNet/ZeroMail#20]

**Comment by HelloZeroNet** [comment]
Unfortunetly with the current working method it's not possible, because you need the display logic to decrypt the messages

**Comment by Plasmmer** [comment]
Possible when web workers come to ZeroNet (sites working in background). Like BitMessage does.

**Comment by Emeraude** [comment]
Are the metadata also encrypted ? If not we could just display notifications like "You got 1 new message today", without any more information.

**Comment by HelloZeroNet** [comment]
Yes everything is encrypted, so it's not possible right now

**Comment by amuuza** [comment]
What a pity there is no notifier. It is impressive to see a serverless asynchronous messaging tool working like a charm.

**Comment by stbinan** [comment]
Here you go https://git.gnu.io/chbinnc/binchan-s-messy-repo/tree/master/ZeroMailBot
> a Python2 script that decrypts your ZeroMail and sends them to a local bot. After you schedule the bot and follow it, you can receive new ZeroMail notification from ZeroHello newsfeed.

And according to shortcutme, it needs a lot of efforts to make it into a ZeroNet plugin:
> Currently the site's notification feed is defined by a sql request. To make zeromail notifications possible we need to allow javascript logic to return the latest message list.
It could be possible by extending the newsfeed plugin to support javascript function call signups. After that when the user visits ZeroHello, the plugin would create a sandboxed iframe to load and run the newsfeed returning javascript function and then inserts the items returned to the newsfeed.

**Comment by lezsakdomi** [comment]
Yes, this is an essential feature. I have two more solution ideas:

* *Update hook*
  Run a script after a zite update (specified in `content.json`).
  This causes possible hangs after updates, so I think this idea should be rejected.
  _Unless_ one develops an easy-to-write and forcefully deterministic language for the purpose.
  Needs change in HelloZero/ZeroNet

* *On-query decryption*
  Provide more scripting oppurtunity feedFollow
  Yes, a single query should be enough, but with some more advanced SQL functions (decrypt(), getUserAddress()).
  This needs an external change too.

* *SiteWorker*
  See previos comment.
  Since the reference refers to ZeroNetPlus, which is "just" a fork, I think we shouldn't rely on that globally.

I think at least on solution should be implemented. The problem this site is facing could arise by the development of other sites too.

**Comment by DaniellMesquita** [comment]
@lezsakdomi

There is already [BackgroundProcessing Plugin](https://github.com/HelloZeroNet/Plugin-BackgroundProcessing), made by @imachug, that can power this notifications feature for encrypted data
